bin/prepare_entity_binding_ratios.py

```diff
@@ -53,21 +56,51 @@ def call_binder(score, method):
-    protein_peptide_occs      = pd.read_csv(args.protein_peptide_occ, sep='\t').drop(columns="count")
-    entities_proteins_occs    = pd.read_csv(args.entities_proteins_occ, sep='\t')
-    microbiomes_entities_occs = pd.read_csv(args.microbiomes_entities_occ, sep='\t')
-    conditions                = pd.read_csv(args.conditions, sep='\t')
-    condition_allele_map      = pd.read_csv(args.condition_allele_map, sep='\t')
-    alleles                   = pd.read_csv(args.alleles, sep='\t')
+    predictions                     = pd.read_csv(args.predictions, sep='\t', index_col="peptide_id").sort_index()
+    predictions["allele_id"]        = pd.to_numeric(predictions["allele_id"], downcast="unsigned")
+    predictions["prediction_score"] = pd.to_numeric(predictions["prediction_score"], downcast="float")
+    # NOTE could be read in chunk-wise if this gets bottleneck.
+    # E.g. sort and split before by peptide_ids into chunks, read in as file list, process chunk-wise (protein_peptide_occs accordingly)
+    # (would decrease code readability though)
+    protein_peptide_occs               = pd.read_csv(args.protein_peptide_occ, sep='\t', index_col="peptide_id").sort_index()
+    protein_peptide_occs["protein_id"] = pd.to_numeric(protein_peptide_occs["protein_id"], downcast="unsigned")
+    protein_peptide_occs["count"]      = pd.to_numeric(protein_peptide_occs["count"], downcast="unsigned")
+
+    entities_proteins_occs               = pd.read_csv(args.entities_proteins_occ, sep='\t')
+    entities_proteins_occs["entity_id"]  = pd.to_numeric(entities_proteins_occs["entity_id"], downcast="unsigned")
+    entities_proteins_occs["protein_id"] = pd.to_numeric(entities_proteins_occs["protein_id"], downcast="unsigned")
+
+    microbiomes_entities_occs                  = pd.read_csv(args.microbiomes_entities_occ, sep='\t')
+    microbiomes_entities_occs["microbiome_id"] = pd.to_numeric(microbiomes_entities_occs["microbiome_id"], downcast="unsigned")
+    microbiomes_entities_occs["entity_id"]     = pd.to_numeric(microbiomes_entities_occs["entity_id"], downcast="unsigned")
+    microbiomes_entities_occs["entity_weight"] = pd.to_numeric(microbiomes_entities_occs["entity_weight"], downcast="float")
+
+    conditions                  = pd.read_csv(args.conditions, sep='\t')
+    conditions["condition_id"]  = pd.to_numeric(conditions["condition_id"], downcast="unsigned")
+    conditions["microbiome_id"] = pd.to_numeric(conditions["microbiome_id"], downcast="unsigned")
+
+    condition_allele_map                 = pd.read_csv(args.condition_allele_map, sep='\t')
+    condition_allele_map["condition_id"] = pd.to_numeric(condition_allele_map["condition_id"], downcast="unsigned")
+    condition_allele_map["allele_id"]    = pd.to_numeric(condition_allele_map["allele_id"], downcast="unsigned")
+
+    alleles              = pd.read_csv(args.alleles, sep='\t')
+    alleles["allele_id"] = pd.to_numeric(alleles["allele_id"], downcast="unsigned")
+
```

Comment:
I wonder if it makes a difference to use downcast to int for allele_id and condition_id as I believe it should be enough resources. Does it make a difference at all? For sure compared to no downcast but between unsigned and int. If it makes a difference it may be useful to check the largest id and decide on the fly which type to use?

Reply:
Why would it be better to downcast to `int`? For the same number of bytes you can cover a larger range with `unsigned`.

Reply:
Well sorry for that I misread the memory usage of these types, I should read everything and not just the maximum of a number. True it will not change anything!